java-frontend/src/main/java/org/sonar/java/se/checks/SECheck.java

```diff
@@ -75,6 +80,36 @@ public void reportIssue(Tree tree, String message, Set<List<JavaFileScannerConte
       .orElse(new SEIssue(tree, message, secondary)));
   }
 
+  public static List<JavaFileScannerContext.Location> flow(ExplodedGraph.Node currentNode, SymbolicValue currentVal ) {
+    List<JavaFileScannerContext.Location> flow = new ArrayList<>();
+    ExplodedGraph.Node node = currentNode;
+    Symbol lastEvaluated = currentNode.programState.getLastEvaluated();
+    while (node != null) {
+      ExplodedGraph.Node finalNode = node;
+      if(finalNode.programPoint.syntaxTree() != null) {
```

Comment:
this is perhaps not related to your change, but I think this loop will be more readable if conditions is inverted with `continue`